include/psa/crypto_extra.h

```diff
@@ -1901,6 +1901,331 @@ static inline struct psa_pake_operation_s psa_pake_operation_init( void )
+ * \retval #PSA_ERROR_STORAGE_FAILURE
+ * \retval #PSA_ERROR_DATA_CORRUPT
+ * \retval #PSA_ERROR_DATA_INVALID
+ * \retval #PSA_ERROR_INSUFFICIENT_ENTROPY
+ * \retval #PSA_ERROR_BAD_STATE
+ *         The library has not been previously initialized by psa_crypto_init().
+ *         It is implementation-dependent whether a failure to initialize
+ *         results in this error code.
+ */
+int psa_sign_hash_complete( psa_sign_interruptible_context *ctx,
+                            unsigned char *signature, size_t signature_size,
+                            size_t *signature_len );
+
+/**
+ * \brief                       Start reading and verifying a hash or short
+ *                              message, in an interruptible manner.
+ *
+ * \see                         \c psa_verify_hash_complete()
+ *
+ * \note                        This function combined with \c
+ *                              psa_verify_hash_complete() is equivalent to
+ *                              \c psa_verify_hash() but it can return early
+ *                              and restart according to the limit set with \c
+ *                              psa_sign_interruptible_set_max_ops() to reduce
+ *                              blocking.
+ *
+ * \note                        Users should call \c psa_verify_hash_complete()
+ *                              on the same context after a successful call to
+ *                              this function this until \c
+ *                              psa_verify_hash_complete() either returns 0 or
+ *                              an error. \c psa_verify_hash_complete() will
+ *                              return #PSA_ERROR_IN_PROGRESS if there is more
+ *                              work to do.
+ *
+ * \param ctx                   The \c psa_sign_interruptible_context to use.
+ *                              This must be initialized first.
+ *
+ * \param key                   Identifier of the key to use
+ *                              for the operation. It must be an asymmetric key
+ *                              pair. The key must allow the usage
+ *                              #PSA_KEY_USAGE_VERIFY_HASH.
+ * \param alg                   A signature algorithm (PSA_ALG_XXX
+ *                              value such that #PSA_ALG_IS_SIGN_HASH(\p alg)
+ *                              is true), that is compatible with
+ *                              the type of \p key.
+ * \param[in] hash              The hash or message whose signature is to be
+ *                              verified.
+ * \param hash_length           Size of the \p hash buffer in bytes.
+ * \param[in] signature         Buffer containing the signature to verify.
+ * \param signature_length      Size of the \p signature buffer in bytes.
+ *
+ * \retval #PSA_SUCCESS
+ *         The operation started successfully - please call \c
+ *         psa_verify_hash_complete() to complete the operation()
+ *
+ * \retval #PSA_ERROR_NOT_SUPPORTED
```

Comment:
Should include PSA_ERROR_IN_PROGRESS here

Reply:
I was not thinking the start function calls could return anything other than success or error - its the complete functions that are splittable in my mind.